src/ray/core_worker/transport/actor_task_submitter.cc

```diff
@@ -379,9 +379,6 @@ void ActorTaskSubmitter::DisconnectActor(const ActorID &actor_id,
     absl::MutexLock lock(&mu_);
     auto queue = client_queues_.find(actor_id);
     RAY_CHECK(queue != client_queues_.end());
-    if (!dead) {
-      RAY_CHECK_GT(num_restarts, 0);
```

Comment:
Since we don't increase num_restarts for preemption, this check is no longer true.

Reply:
Q: don't we lose some observability information: we don't know how many restarts are due to preemptions.

Reply:
Agree. I think we should expose to the state API: we should add to

```
@dataclass(init=not IS_PYDANTIC_2)
class ActorState(StateSchema):
    """Actor State"""
```

We are already exposing the number of restarts due to lineage reconstruction there.

Reply:
Added.

Reply:
I have changed back to the approach of incrementing both num_restarts and num_restarts_due_to_node_preemption. num_restarts_due_to_node_preemption is now exposed to ActorState and also logged out.